package NYT.NChunkClient.NProto;

option python_package = "yt_proto.yt.client.chunk_client.proto";

////////////////////////////////////////////////////////////////////////////////

// A boundary can be given by:
// * row index (for tables and journals only)
// * chunk index
// * offset (for files only)
// * key (for tables only)
// A boundary is either inclusive or exclusive depending of the context.
message TReadLimit
{
    optional int64 row_index = 1;
    optional int32 chunk_index = 2;
    optional int64 offset = 3;
    optional bytes key = 4;

    // Sometimes keys in read limis are transported in wire format via attachments.
    // This index indicates the proper key inside attachments' rowset.
    optional int32 key_index = 5;
}

message TReadRange
{
    optional TReadLimit lower_limit = 1;
    optional TReadLimit upper_limit = 2;
}

////////////////////////////////////////////////////////////////////////////////
